screens/QrScan/QrScanScreen.tsx

```diff
@@ -21,20 +30,32 @@ const QrScanScreen = () => {
     }
   }, [isFocused])
 
-  const handleBarCodeScanned = ({ type, data }: BarCodeScanningResult): void => {
+  const handleUser = async (id : string) => {
+    if(id === userId){
+      showErrorToast("This is you !")
+    } else {
+      const scannedUser = await getUserData(id)
+      if(scannedUser === null || scannedUser === undefined){
+        showErrorToast("User not found")
+      } else {
+        navigation.navigate("AddContact", {uid: id})
+      }
+    }
+  }
+
+  const handleEvent = async (id : string) => {
+    showErrorToast("events QR codes not implemented yet, event id: " + id)
+  }
+
+  const handleBarCodeScanned = ({ data }: BarCodeScanningResult): void => {
     if(!qrScanned){
       setQrScanned(true)
-      Alert.alert(
-        "QR scanned", 
-        `type: ${type}\ndata:${data}`,
-        [
-          { 
-            text: "OK", 
-            onPress: () => setQrScanned(false)
-          },
-        ],
-        { cancelable: false }
-      )
+      const path = data.replace(Linking.createURL("/"), "")
+      const [route, id] = path.split('/')
+      if(route === "contact") handleUser(id)
+      else if (route === "event") handleEvent(id)
+      else showErrorToast("Qr code not recognized")
+      setTimeout(() => setQrScanned(false), 1500) // re-allow to scan QR in 1.5 second
```

Comment:
Might be worth using a debouncer here: https://usehooks.com/usedebounce

Reply:
What about this useDebounce: https://www.npmjs.com/package/use-debounce ? I'm asking you because I think you'll implement the debouncer for the search bar so it would be good if we use the same.

Reply:
Any will do, we could even build the hook from scratch, it's a very simple logic! I'll follow whichever you pick. Yours seems better!

Reply:
Ok I'll take this one then: https://www.npmjs.com/package/use-debounce